src/app-layout/index.tsx

```diff
@@ -218,10 +276,13 @@ const OldAppLayout = React.forwardRef(
 
     const closedDrawerWidth = 40;
     const effectiveNavigationWidth = navigationHide ? 0 : navigationOpen ? navigationWidth : closedDrawerWidth;
+
     const effectiveToolsWidth =
-      toolsHide && (!splitPanelDisplayed || splitPanelPreferences?.position !== 'side')
+      toolsHide && (!splitPanelDisplayed || splitPanelPreferences?.position !== 'side') && !drawers
         ? 0
-        : toolsOpen
+        : selectedDrawer?.resizable
+        ? drawerSize
+        : toolsOpen || activeDrawerId
         ? toolsWidth
         : closedDrawerWidth;
```

Comment:
Having that many members in this condition exceeds my ability to understand the code. Let's extract this into a function, at least (and possibly simplify some branches)

Reply:
Makes sense.